test/end2end_test.go

```diff
@@ -635,13 +658,24 @@ func (te *test) listenAndServe(ts testpb.TestServiceServer, listen func(network,
 	}
 	sopts = append(sopts, te.customServerOptions...)
 	s := grpc.NewServer(sopts...)
-	te.srv = s
-	if te.healthServer != nil {
-		healthgrpc.RegisterHealthServer(s, te.healthServer)
+
+	// Create a new default health server if enableHealthServer is set, or use
+	// the provided one.
+	var hs healthpb.HealthServer
+	if ts != nil {
+		if tss, ok := ts.(*testServer); ok {
+			if tss.enableHealthServer {
```

Comment:
Why is `enableHealthServer` a field of `testServer`, not field of `test`?
If it's a field of test, you wouldn't need this type assertion.

Reply:
Done. Thanks.